Library/Homebrew/cmd/man.rb

```diff
@@ -25,21 +25,31 @@ def man
       puts "Writing HTML fragments to #{DOC_PATH}"
       puts "Writing manpages to #{TARGET_PATH}"
 
-      Dir["#{SOURCE_PATH}/*.md"].each do |source_file|
-        args = %W[
-          --pipe
-          --organization=Homebrew
-          --manual=brew
-          #{source_file}
-        ]
-        page = File.basename(source_file, ".md")
-
-        target_html = DOC_PATH/"#{page}.html"
-        target_html.atomic_write Utils.popen_read("ronn", "--fragment", *args)
-
-        target_man = TARGET_PATH/page
-        target_man.atomic_write Utils.popen_read("ronn", "--roff", *args)
-      end
+      header = IO.read("#{SOURCE_PATH}/header.1.md")
+      footer = IO.read("#{SOURCE_PATH}/footer.1.md")
+      sub_commands = Dir["#{HOMEBREW_REPOSITORY}/Library/Homebrew/cmd/*"].sort.map { |source_file|
+        IO.read(source_file).
+          split("\n").
+          grep(/^#:/).
+          map { |line| line.slice(2..-1) }.
+          join("\n")
+      }.reject { |s| s.strip.empty? }.join("\n\n")
```

Comment:
A bunch of things regarding this block:

- I think we can safely shorten `sub_commands` to `commands`.
- Make better use of the power of `Pathname`:
  - `Pathname.glob("#{HOMEBREW_LIBRARY_PATH}/cmd/*.{rb,sh}")` instead of `Dir[]` (also use `HOMEBREW_LIBRARY_PATH` instead of `HOMEBREW_REPOSITORY` for brevity; more explicit about which files we want to inspect).
  - With this, `IO.read(source_file)` can become `source_file.read`.
- For multi-line blocks we prefer `begin`/`end` instead of curly braces.
- Personally, I think it doesn't hurt to split this block into multiple expressions, even if that means assigning the result multiple times. There's a tad bit too much going on in this expression.
- I think we should be sorting via `sort_by { |source_file| source_file.basename.sub(/\.(rb|sh)$/, "") }`, i.e. disregarding the file extension. Otherwise `tap-info` comes before `tap` and it should be the other way round.